mm2src/mm2_main/src/ext_api/one_inch/rpcs.rs

```diff
@@ -0,0 +1,437 @@
+                                       TokensResponse};
+
+/// "1inch_v6_0_classic_swap_contract" rpc impl
+/// used to get contract address (for e.g. to approve funds)
+pub async fn one_inch_v6_0_classic_swap_contract_rpc(
+    _ctx: MmArc,
+    _req: AggregationContractRequest,
+) -> MmResult<String, ApiIntegrationRpcError> {
+    Ok(ApiClient::classic_swap_contract().to_owned())
+}
+
+/// "1inch_classic_swap_quote" rpc impl
+pub async fn one_inch_v6_0_classic_swap_quote_rpc(
+    ctx: MmArc,
+    req: ClassicSwapQuoteRequest,
+) -> MmResult<ClassicSwapResponse, ApiIntegrationRpcError> {
+    let (base, base_contract) = get_coin_for_one_inch(&ctx, &req.base).await?;
+    api_supports_coin(&base)?;
+    let (rel, rel_contract) = get_coin_for_one_inch(&ctx, &req.rel).await?;
+    let sell_amount = wei_from_big_decimal(&req.amount.to_decimal(), base.decimals())
+        .mm_err(|err| ApiIntegrationRpcError::InvalidParam(err.to_string()))?;
+    let query_params = ClassicSwapQuoteParams::new(base_contract, rel_contract, sell_amount.to_string())
+        .with_fee(req.fee)
+        .with_protocols(req.protocols)
+        .with_gas_price(req.gas_price)
+        .with_complexity_level(req.complexity_level)
+        .with_parts(req.parts)
+        .with_main_route_parts(req.main_route_parts)
+        .with_gas_limit(req.gas_limit)
+        .with_include_tokens_info(Some(req.include_tokens_info))
+        .with_include_protocols(Some(req.include_protocols))
+        .with_include_gas(Some(req.include_gas))
+        .with_connector_tokens(req.connector_tokens)
+        .build_query_params()
+        .mm_err(|api_err| ApiIntegrationRpcError::from_api_error(api_err, Some(base.decimals())))?;
+    let quote = ApiClient::new(ctx)
+        .mm_err(|api_err| ApiIntegrationRpcError::from_api_error(api_err, Some(base.decimals())))?
+        .call_swap_api(
+            base.chain_id(),
+            ApiClient::get_quote_method().to_owned(),
+            Some(query_params),
+        )
+        .await
+        .mm_err(|api_err| ApiIntegrationRpcError::from_api_error(api_err, Some(base.decimals())))?; // use 'base' as amount in errors is in the src coin
+    ClassicSwapResponse::from_api_classic_swap_data(quote, rel.decimals()) // use 'rel' as quote value is in the dst coin
+        .mm_err(|err| ApiIntegrationRpcError::ApiDataError(err.to_string()))
+}
+
+/// "1inch_classic_swap_create" rpc implementation
+/// This rpc actually returns a transaction to call the 1inch swap aggregation contract. GUI should sign it and send to the chain.
+/// We don't verify the transaction in any way and trust the 1inch api.
+pub async fn one_inch_v6_0_classic_swap_create_rpc(
+    ctx: MmArc,
+    req: ClassicSwapCreateRequest,
+) -> MmResult<ClassicSwapResponse, ApiIntegrationRpcError> {
+    let (base, base_contract) = get_coin_for_one_inch(&ctx, &req.base).await?;
+    api_supports_coin(&base)?;
```

Comment:
Same here

Reply:
Added check for same chain to api_supports_coin() in [65c4853](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2222/commits/65c4853fd1c34ef8c3527a9becc31e04b12dcd38)